sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -510,8 +510,13 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     requireTableExists(db, tableDefinition.identifier.table)
     verifyTableProperties(tableDefinition)
 
+    // Add table metadata such as table schema, partition columns, etc. if they aren't already
+    // present.
+    val withMetaProps = tableDefinition.copy(
```

Comment:
Sounds reasonable to me. This would be a new public method only for ```HiveExternalCatalog```, correct (i.e. it shouldn't be added to the ```ExternalCatalog``` interface)?